services/sts/src/main/java/software/amazon/awssdk/services/sts/auth/StsWebIdentityTokenFileCredentialsProvider.java

```diff
@@ -0,0 +1,336 @@
+        }
+
+
+        /**
+         * Configure the {@link AssumeRoleWithWebIdentityRequest} that should be periodically sent to the STS service to update
+         * the session token when it gets close to expiring.
+         *
+         * @param assumeRoleWithWebIdentityRequest The request to send to STS whenever the assumed session expires.
+         * @return This object for chained calls.
+         */
+        public Builder refreshRequest(AssumeRoleWithWebIdentityRequest assumeRoleWithWebIdentityRequest) {
+            return refreshRequest(() -> assumeRoleWithWebIdentityRequest);
+        }
+
+        /**
+         * Similar to {@link #refreshRequest(AssumeRoleWithWebIdentityRequest)}, but takes a {@link Supplier} to supply the
+         * request to STS.
+         *
+         * @param assumeRoleWithWebIdentityRequestSupplier A supplier
+         * @return This object for chained calls.
+         */
+        public Builder refreshRequest(Supplier<AssumeRoleWithWebIdentityRequest> assumeRoleWithWebIdentityRequestSupplier) {
+            this.assumeRoleWithWebIdentityRequestSupplier = assumeRoleWithWebIdentityRequestSupplier;
+            return this;
+        }
+
+        /**
+         * Similar to {@link #refreshRequest(AssumeRoleWithWebIdentityRequest)}, but takes a lambda to configure a new {@link
+         * AssumeRoleWithWebIdentityRequest.Builder}. This removes the need to call {@link
+         * AssumeRoleWithWebIdentityRequest#builder()} and {@link AssumeRoleWithWebIdentityRequest.Builder#build()}.
+         */
+        public Builder refreshRequest(Consumer<AssumeRoleWithWebIdentityRequest.Builder> assumeRoleWithWebIdentityRequest) {
+            return refreshRequest(AssumeRoleWithWebIdentityRequest.builder().applyMutation(assumeRoleWithWebIdentityRequest)
+                                                                  .build());
+        }
+
+        @Override
+        public StsWebIdentityTokenFileCredentialsProvider build() {
+            if (this.stsClient == null) {
+                OrRetryCondition retryCondition =
+                    OrRetryCondition.create(context -> context.exception() instanceof IdpCommunicationErrorException,
+                                            RetryCondition.defaultRetryCondition());
+                StsClient localStsClient = StsClient.builder()
+                                                    .applyMutation(this::configureEndpoint)
+                                                    .credentialsProvider(AnonymousCredentialsProvider.create())
+                                                    .overrideConfiguration(
+                                                        o -> o.retryPolicy(r -> r.retryCondition(retryCondition)))
+                                                    .build();
+
+                this.stsClient = localStsClient;
+                super.stsClient(localStsClient);
+                isStsClientDefault = true;
```

Comment:
As discussed offline, it might be better to always require an StsClient to be consistent with other StsCredentialProviders

Reply:
done